base_multi_company/models/multi_company_abstract.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, fields, models
+
+
+class MultiCompanyAbstract(models.AbstractModel):
+
+    _name = 'multi.company.abstract'
+    _description = 'Multi-Company Abstract'
+
+    company_id = fields.Many2one(
+        string='Company',
+        comodel_name='res.company',
+        compute='_compute_company_id',
+        store=True,
```

Comment:
@pedrobaeza @lasley  my main concern here is that the stored value is not predictable and not right for all users (always the last one into the list of available company). Now that it's possible to all the users to see the list of companies (company_ids -> 'res.company.assignment')  where the product is available even if they are not part of one of these companies, the value stored into this company_id field could be one company not allowed for the current user.... IMO this field should definitely never be stored to avoid unpredictable side effect